packages/odata-common/src/request/odata-request-config.ts

```diff
@@ -38,6 +38,8 @@ export abstract class ODataRequestConfig {
   private _appendedPaths: string[] = [];
   private _fetchCsrfToken = true;
   private _timeout: number | undefined = undefined;
+  private _contentIdChangesetHeader: string | undefined;
+  private _contentIdChangesetUrl: string | undefined;
```

Comment:
[pp] I feel `Changeset` is not necessary, as you can use the header/url for both batch operations (retrieve/changeset). Maybe use `_contentIdHeader` and `_contentIdChangesetUrl` ?

[fyi]
1. As we want to support content id in the request, we might also want to think about the [response](https://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_MultipartBatchResponse).

2. In addition, the `contentId` can also be used with etag.
[reference](https://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_ReferencinganETag)

[pp] Given the batch content id 3 use cases, I would do the following:
```
interface BatchContentId{
  header?: string | undefined;
  url?: string | undefined;
  etag?: string | undefined;
}
OdataRequestConfig{
  private _batchContentId: BatchContentId | undefined;
}
```

Reply:
oh, I missed the `eTag` thing. You are right with three facets a object makes total sense. I adjust this draft.